Add get_docval_macro method

rly opened the pull request:
## Motivation

Add a method to get a tuple of the current values for a docval_macro, e.g., 'array_data' and 'scalar_data'. This is generally useful and specifically useful for working with the allowed types where a docval macro is used to restrict the argument types.

## How to test the behavior?
```python
from hdmf.utils import get_docval_macro
get_docval_macro('array_data')
```

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/hdmf-dev/hdmf/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

rly committed:
Add get_docval_macro method

rly committed:
Update changelog

oruebel commented:
FYI, I added a variant of that also in https://github.com/hdmf-dev/hdmf/pull/98 The main difference is that I had implemented this as a hidden function and I also need to be able to get the full dict, not just a single item. We can update the Zarr PR to use this PR to get this in first, but it would be great to have the feature for ``key=None`` to get a copy of the full array as well, as I need it at least for implementing tests. 

```
def __get_docval_macros(key=None):
    """
    Return a deepcopy of __macros. Used in the test-suite to avoid hardcoding in the tests
    """
    if key is None:
        return _copy.deepcopy(__macros)
    else:
        return __macros[key]
```

oruebel commented:
Can we just add the ``macro=None`` case directly in this PR then I won't have to change in the other PR and just need to update the other places in the Zarr PR to use this function instead.

rly committed:
Update get_docval_macro, add 'data' to macros

rly commented:
Sorry about the early request for review. I clicked it before reading your first comment.

I updated the `get_docval_macro` function to be basically the same as your function. 

I also added the macro `'data': []` to help show that this is used. When the `DataIO` class is loaded, it is added to the 'data' macro.

rly committed:
Update test

rly committed:
Merge branch 'dev' into get_docval_macro

oruebel commented:
I added a note to the TODO list in the Zarr PR to make sure I replace the function there with this one, to avoid accidentally creating two. This PR looks good to me.
